src/features/DisplayControls/DrugChart/components/CalendarHeader.jsx

```diff
@@ -11,15 +11,16 @@ export default function CalendarHeader(props) {
   return (
     <div className="calendar-header">
       <div style={{ display: "flex" }}>
-        {currentShiftArray.map((hour) => {
-          const transformedHour = enable24HourTime ? hour : hour % 12 || 12;
+        {currentShiftArray.map((time) => {
+          const [hour, minute] = time.split(":");
+          const transformedHour = enable24HourTime ? hour : (hour % 12) || 12;
           const period = enable24HourTime ? "" : hour < 12 ? "AM" : "PM";
           const formattedHour = `${transformedHour.toLocaleString("en-US", {
             minimumIntegerDigits: 2,
-          })}:00 ${period}`;
+          })}:${minute} ${period}`;
```

Comment:
can we refactor this and use utility methods from DateTimeUtils?

Reply:
I looked into the utility methods, didn't find a suitable one that can be used here. Should I create a new one an use it over here?

Reply:
okay, I though we had, if not lets use moment instead of toLocaleString and having hour < 12 for checking AM or PM. It would be great if you would refactor and use moment here.